### DISCUSSION TITLE
BulkAction hidden / visible closure.

### DISCUSSION THREAD

**Discussion by RSickenberg** [filamentphp/filament#13765] [Help]
### Package

Actions

### Package Version

v3.2.97

### How can we help you?

Basically trying to create a bulk action but I want them to be conditionally visible on my table.

Here is my implementation:
```php
    #[Override]
    public static function bulk(): BulkAction
    {
        return BulkAction::make('activate')
            ->icon('heroicon-o-eye')
            ->color('info')
            ->fetchSelectedRecords()
            ->deselectRecordsAfterCompletion()
            ->disabled(static fn (Flight $record) => ! \Gate::check('transitionToVisible', $record))
            ->action(static fn (Collection $records) => self::execute($records, $records->count() > 1))
            ->label(__('flights.actions.activate'));
    }
```

I tried with `->visible(fn (?Collection $records) => $records->each->dosmth),` or the `->hidden` as well, but I'm not able to have something else than records on page load / after selection.

**Comment by dissto** [comment]
Are you trying to [disable bulk actions for some rows](https://filamentphp.com/docs/3.x/tables/actions#disabling-bulk-actions-for-some-rows) ?

**Reply by RSickenberg** [reply]
That's a negative @dissto, but good question.
